src/main/java/com/commercetools/pingpong/service/impl/GameServiceImpl.java

```diff
@@ -13,18 +13,19 @@
     public GameServiceImpl() {
         this.match = new Game(new Player("HeshamOne"), new Player("HeshamTwo"));
     }
+    // FIXME: These hardcoded values will be removed later
 
     @Override
     public void updateScore(final Message message) {
         Player actingPlayer = getPlayer(message);
-        match.updateScoreOfPlayer(actingPlayer, message.getActionType());
-
-        if (match.hasSomebodyWon()) {
+        if (match.areBothSetScoresZero() && !match.isSomeoneServing()) {
+            match.setFirstServe(actingPlayer);
+        } else {
+            match.updateScoreOfPlayer(actingPlayer, message.getActionType());
             match.updateMatchScoreOfPlayers();
+            match.isGameOver();
```

Comment:
From the name of the function one would think the function returns a Boolean indicating whether the game is over or not. But behind the sense this function has a lot of unexpected side-effects like resetting scores and serves. This should probably be better indicated.

Reply:
Yes, i think that's better
7f478b8676bb69b6b7de7cba6f3d42fbbb92ffdb